src/app.ts

```diff
@@ -54,6 +61,24 @@ async function main() {
     },
     defaultErrorHandler: false,
   });
+
+  const entityManager = getManager();
+  const reportService = new ReportService(entityManager);
+  const reportController = new ReportController(reportService);
+
+  app.set('view engine', 'pug')
+
+  app.get('/reports/admin/', async (req: UserModel, res: any) => {
+    const user = req as UserModel;
+    user.admin = true;
+    const postReports = await reportController.getAllPostReports(user);
+    const profileReports = await reportController.getAllProfileReports(user);
+    const messageReports = await reportController.getAllMessageReports(user);
+    res.render('admin', { 
+      postReports: reportToString(postReports), 
+      profileReports: reportToString(profileReports), 
+      messageReports: reportToString(messageReports) });
+  });
```

Comment:
Am I misunderstanding something or is this route not protected? It seems like anyone who sends a get request to this route with user data in the request body will become an admin user and then be able to view all reports -- we should have some sort of middleware to protect routes like this for admin users -- same goes for all the routes in the reports controller

Reply:
Yes, this route is unprotected as of now. I didn't know how to add a middleware to protect it

Reply:
I figured out how to protect it. Frontend will be sent an html response that they can then format. Is that good?

Reply:
Yup makes sense